io.openems.edge.meter.siemens.pac1600/src/io/openems/edge/meter/siemens/pac1600/MeterSiemensPAC1600Impl.java

```diff
@@ -0,0 +1,166 @@
+					m(ElectricityMeter.ChannelId.ACTIVE_POWER_L2, new SignedDoublewordElement(21),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.ACTIVE_POWER_L3, new SignedDoublewordElement(23),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					
+					m(ElectricityMeter.ChannelId.REACTIVE_POWER_L1, new SignedDoublewordElement(25),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.REACTIVE_POWER_L2, new SignedDoublewordElement(27),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.REACTIVE_POWER_L3, new SignedDoublewordElement(29),ElementToChannelConverter.SCALE_FACTOR_MINUS_2)),
+				
+				new FC3ReadRegistersTask(49, Priority.HIGH, // DummyRegisterElement didn`t work here, there are only 30 Registers in one Task allowed
+					m(ElectricityMeter.ChannelId.FREQUENCY, new SignedDoublewordElement(49),ElementToChannelConverter.SCALE_FACTOR_1),
+					m(ElectricityMeter.ChannelId.VOLTAGE, new SignedDoublewordElement(51),ElementToChannelConverter.SCALE_FACTOR_1),
+					new DummyRegisterElement(53, 56), //
+					m(ElectricityMeter.ChannelId.ACTIVE_POWER, new SignedDoublewordElement(57),ElementToChannelConverter.SCALE_FACTOR_MINUS_2),
+					m(ElectricityMeter.ChannelId.REACTIVE_POWER, new SignedDoublewordElement(59),ElementToChannelConverter.SCALE_FACTOR_MINUS_2)),
+					
+				new FC3ReadRegistersTask(6687, Priority.HIGH,
+					m(ElectricityMeter.ChannelId.ACTIVE_CONSUMPTION_ENERGY, new UnsignedDoublewordElement(6687),ElementToChannelConverter.DIRECT_1_TO_1),
+					m(ElectricityMeter.ChannelId.ACTIVE_PRODUCTION_ENERGY, new UnsignedDoublewordElement(6689),ElementToChannelConverter.DIRECT_1_TO_1),
+					new DummyRegisterElement(6691, 6706),
+					m(ElectricityMeter.ChannelId.ACTIVE_CONSUMPTION_ENERGY_L1, new UnsignedDoublewordElement(6707),ElementToChannelConverter.DIRECT_1_TO_1),
+					m(ElectricityMeter.ChannelId.ACTIVE_PRODUCTION_ENERGY_L1, new UnsignedDoublewordElement(6709),ElementToChannelConverter.DIRECT_1_TO_1)),
+				
+				new FC3ReadRegistersTask(6727, Priority.HIGH, // DummyRegisterElement didn`t work here, there are only 30 Registers in one Task allowed
+					m(ElectricityMeter.ChannelId.ACTIVE_CONSUMPTION_ENERGY_L2, new UnsignedDoublewordElement(6727),ElementToChannelConverter.DIRECT_1_TO_1),
+					m(ElectricityMeter.ChannelId.ACTIVE_PRODUCTION_ENERGY_L2, new UnsignedDoublewordElement(6729),ElementToChannelConverter.DIRECT_1_TO_1),
+					new DummyRegisterElement(6731, 6746),
+					m(ElectricityMeter.ChannelId.ACTIVE_CONSUMPTION_ENERGY_L3, new UnsignedDoublewordElement(6747),ElementToChannelConverter.DIRECT_1_TO_1),
+					m(ElectricityMeter.ChannelId.ACTIVE_PRODUCTION_ENERGY_L3, new UnsignedDoublewordElement(6749),ElementToChannelConverter.DIRECT_1_TO_1))); 
+		
+		// Calculates required Channels from other existing Channels.
+		this.addCalculateChannelListeners();
```

Comment:
I wouldn't do this here, but in the `activate` method. Furthermore, one can use the static method `ElectricityMeter.calculateSumCurrentFromPhases(this);`